app/controllers/UpscanController.scala

```diff
@@ -36,15 +36,32 @@ import scala.concurrent.{ExecutionContext, Future}
 class UpscanController @Inject()(authAction: AuthAction,
                                  sessionService: SessionService,
                                  mcc: MessagesControllerComponents,
-                                 override val global_error: views.html.global_error
+                                 override val global_error: views.html.global_error,
+                                 fileUploadProblemView: views.html.file_upload_problem
                                 )(implicit executionContext: ExecutionContext, ersUtil: ERSUtil,
                                   actorSystem: ActorSystem, override val appConfig: ApplicationConfig)
   extends FrontendController(mcc) with Retryable with I18nSupport with BaseController {
 
   val logger: Logger = Logger(getClass)
+
+
   def failure(): Action[AnyContent] = authAction.async { implicit request =>
```

Comment:
In terms of unit testing again, I think we should cover:
1. Valid request which matches the first case
2. Valid request which matches the second case
3. Invalid request

Reply:
Done!